### DISCUSSION TITLE
FEATURE: Specialist Knowledge Integration

### DISCUSSION THREAD

**Discussion by colin-stannard** [LAION-AI/Open-Assistant#3399] [Ideas]
One of the most powerful benefits of self-hosting an AI assistant would be the ability to extend it with specialist knowledge for your use case.

For example:
 - A company wishing to automate help-and-support for their product with specific / detailed answers to questions about their product or service.
 - A company wishing to retain knowledge about a product they developed (as expertise may leave the company). Effectively documentation that can be queried with questions.

There are many cases where specialist knowledge is also confidential - making self-hosting highly attractive.

I propose the adoption / development of a standardised "plugin" data format that can be used to further train the assistant with custom "specialist knowledge".

This would be accompanied by a self-hosted web UI (perhaps packaged in docker) that would be used by an end-user to develop their own custom scored questions and answers (in much the same way as the current website works).

**Comment by huu4ontocord** [comment]
how do you propose that it would be built? can you give proposed arch and how you would implement it?

**Comment by blankster** [comment]
I don't know how to build it, but that's precisely what my employer is searching too. And exactly also the FAQ. I think this field would have quite some potential, and it's a very common use case.

**Comment by i-am-neo** [comment]
@colin-stannard @blankster Is your knowledge base as is in text format? Something else?

**Comment by pevogam** [comment]
I would relate this issue to this pull request as a specific approach: https://github.com/LAION-AI/Open-Assistant/pull/2765
